### DISCUSSION TITLE
Can I control where node_modules/ is created?

### DISCUSSION THREAD

**Discussion by aaronadamsCA** [yarnpkg/berry#2446] [Q&A] [answered]
Is it possible to make [`yarn workspaces focus`](https://yarnpkg.com/cli/workspaces/focus) create `node_modules/` in the current working directory instead of the workspace root?

Context: I'm using GitHub Actions to deploy function apps from a monorepo to Azure. I've landed on a setup that works well, but it includes a fairly nasty `mv ../../node_modules/ ./` step to move this directory from the workspace root into the package before deployment. This doesn't feel like a very reliable solution.

**Comment by merceyz** [comment, marked as answer]
Not directly but you can use something like this plugin https://gitlab.com/Larry1123/yarn-contrib/-/tree/master/packages/plugin-production-install to make your workspace deployable

**Reply by aaronadamsCA** [reply]
Got it, thank you.

If I understand correctly, it looks like v2 `workspaces focus` always recursively installs all dependencies of all dependent workspaces, whereas v1 `install --focus` installed the dependencies of just the one workspace.

If that's right, that was my gap in understanding; and I suppose it's why v2 `focus` puts `node_modules/` in the workspace root, whereas v1 `install --focus` put it in your package directory.

I can see how that's useful, but it seems unfortunate to have apparently lost the v1 `install --focus` functionality. I'd love to see workspace tools add a flag like `focus --no-recursive` or `--single`, to recreate the v1 functionality of pretending you're in a standalone package.

In the meantime I'll look for another plugin - thank you for the one you suggested, but it's a little more invasive than what I need!

**Reply by merceyz** [reply]
That's right, yes. You can set [`nmHoistingLimits: workspaces`](https://yarnpkg.com/configuration/yarnrc#nmHoistingLimits) to prevent it from hoisting to the root
